src/function/gds/all_shortest_paths.cpp

```diff
@@ -365,7 +379,9 @@ class VarLenAndAllSPOutputWriterPaths : public RJOutputWriter {
                     curPath.pop_back();
                 }
             } else {
-                auto parent = allSpOutputs->bfsGraph.getInitialParentAndNextPtr(topNodeID);
+                auto parent = bfsGraph.getInitialParentAndNextPtr(topNodeID);
+                // TODO(Semih): do we actually need the following while loop?
+                KU_ASSERT(parent->getIter() == top->getIter() - 1);
                 while (parent->getIter() != top->getIter() - 1) {
```

Comment:
Yes we do and I think the ASSERT is wrong. Consider running var-len joins on the following data starting from s: 
s->u->v, s->z1->u, s->z2->u, ..., s->zk->u.

Suppose we are backtracking from v. It will have iter 2, so we will be looking for "iter 1" parent edges of u. However u will first have k iter 2 edges because of the s-z_i->u paths, so we need to iterate through all.

Since all tests pass, it means this case is not covered. Can you add a simple test case for this?

Reply:
Done